suffix_tree/src/lib.rs

```diff
@@ -47,6 +48,7 @@ pub struct SuffixTree<'s> {
     root: Box<Node>,
 }
 
+#[allow(clippy::len_without_is_empty)]
```

Comment:
Am I correct to assume that adding is_empty() would also be incorrect?

From my reading of the code, len==0 doesnt seem to make sense and there's no obvious datastructure to query that would make is_empty() make sense.

I'd gladly implement one if you point it to me.

Otherwise I'll push a commit removing that lint.

Reply:
I haven't touched this code in years. I'm pretty sure nobody is using it. I don't know what makes sense here or not, I'm sorry. Just remove the Clippy `allow`. :)

Reply:
4192c50 fixes this.
I propose changing and refactoring
```
/// The size of the path label into this node.
    pub fn len(&self) -> u32 {
```
into `pub fn path_len` to avoid confusion and remove the lint errors as is,
but if you'd rather keep the code as is, feel free to resolve this conversation here.

Reply:
Please keep it as-is.

Reply:
yup. resolving this here then.